Trimble.Modus.Components/Controls/TMDataGrid/DataGrid.xaml.cs

```diff
@@ -0,0 +1,727 @@
+    /// <param name="unsortedItems">Initial unsortedItems</param>
+    /// <param name="sortData">Contains the column index and sorting order</param>
+    /// <returns></returns>
+    /// <exception cref="NotImplementedException"></exception>
+    private IList<object> GetSortedItems(IEnumerable<object> unsortedItems, SortData sortData)
+    {
+        var columnToSort = Columns[sortData.Index];
+
+        IList<object> items;
+
+        switch (sortData.Order)
+        {
+            case SortingOrder.Ascendant:
+                items = unsortedItems.OrderBy(x => ReflectionUtils.GetValueByPath(x, columnToSort.PropertyName)).ToList();
+                _ = columnToSort.SortingIcon.RotateTo(0, 300, easing: Easing.Linear);
+                break;
+            case SortingOrder.Descendant:
+                items = unsortedItems.OrderByDescending(x => ReflectionUtils.GetValueByPath(x, columnToSort.PropertyName)).ToList();
+                _ = columnToSort.SortingIcon.RotateTo(180, 300, easing: Easing.Linear);
+                break;
+            case SortingOrder.None:
+                items = unsortedItems.ToList();
+                break;
+            default:
+                throw new NotImplementedException();
+        }
+
+        foreach (var column in Columns)
+        {
+            if (column == columnToSort)
+            {
+                column.SortingOrder = sortData.Order;
+                column.SortingIconContainer.IsVisible = sortData.Order != SortingOrder.None;
+            }
+            else
+            {
+                column.SortingOrder = SortingOrder.None;
+                column.SortingIconContainer.IsVisible = false;
+            }
+        }
+
+        return items;
+    }
+
+    private void SortColumn(SortData? sortData = null)
+    {
+        if (ItemsSource is null)
+        {
+            return;
+        }
+
+        sortData ??= SortedColumnIndex;
+
+        var originalItems = ItemsSource.Cast<object>().ToList();
+
+        IList<object> sortedItems;
+
+        if (CanSort(sortData))
+        {
+            sortedItems = GetSortedItems(originalItems, sortData);
+        }
+        else
+        {
+            sortedItems = originalItems;
+        }
+
+
+        InternalItems = sortedItems;
+    }
+    #endregion Sorting methods
+
+    #region Methods
+
+    /// <summary>
+    /// Scrolls to the row
+    /// </summary>
+    /// <param name="item">Item to scroll</param>
+    /// <param name="position">Position of the row in screen</param>
+    /// <param name="animated">animated</param>
+    public void ScrollTo(object item, ScrollToPosition position, bool animated = true) => _collectionView.ScrollTo(item, position: position, animate: animated);
+
+    #endregion Methods
+
+    #region Bindable properties
+    public static readonly BindableProperty ReadOnlyListProperty = BindableProperty.Create(nameof(ReadOnlyList), typeof(List<int>), typeof(DataGrid), defaultValueCreator: _ => new List<int>());
```

Comment:
What is the user of this ReadOnlyList property? I don't see it is used anywhere other than the component itself. Do we need a BindableProperty for this?

Reply:
Guidelines has a "Read Only row" concept, which has a different background color and cannot be selected. 'ReadOnlyList' takes a list of row index as input.
'